{
  "info": {
    "author": "University of Cambridge Information Services",
    "author_email": "devops+fastapi-openapierrors@uis.cam.ac.uk",
    "bugtrack_url": null,
    "classifiers": [
      "Development Status :: 4 - Beta",
      "Intended Audience :: Developers",
      "License :: OSI Approved :: MIT License",
      "Operating System :: OS Independent",
      "Programming Language :: Python :: 3"
    ],
    "description": "# Identity Lib\n\nThis Python package contains utilities enabling nicer error descriptions in\nOpenAPI specs generated by FastAPI.\n\n## Use\n\nInstall `fastapi-openapierrors` using pip:\n\n```\npip install fastapi-openapierrors\n```\n\nThe module can then be used as `fastapi_openapierrors` to annotate routes:\n\n```python3\nfrom fastapi import FastAPI\nfrom fastapi_openapierrors import NotFoundResponse, BadRequestResponse\nfrom pydantic import BaseModel\n\napp = FastAPI()\n\nclass Widget(BaseModel):\n    name: str\n\n@api.get(\n    '/widget/{name}',\n    summary=\"Get a single widget\",\n    description=\"Get a single widget by name\",\n    response_model=Widget,\n    responses={400: BadRequestResponse, 404: NotFoundResponse}\n)\nasync def get_widget(\n      name: str = Path(\n        ...,\n        example=f\"frobnicator2000\",\n        title=\"The name of the widget\"\n      )):\n    # FastAPI automatically returns a Bad Request response if the request does\n    # not match the route.\n\n    # Look for widget by name.\n    widget, was_found = get_widget_by_name(name)\n\n    if not was_found:\n        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND)\n\n    return widget\n```\n\n## Developer quickstart\n\nThis project contains a dockerized testing environment which wraps [tox](https://tox.readthedocs.io/en/latest/).\n\nTests can be run using the `./test.sh` command:\n\n```bash\n# Run all PyTest tests and Flake8 checks\n$ ./test.sh\n\n# Run just PyTest\n$ ./test.sh -e py3\n\n# Run a single test file within PyTest\n$ ./test.sh -e py3 -- tests/test_identifiers.py\n\n# Run a single test file within PyTest with verbose logging\n$ ./test.sh -e py3 -- tests/test_identifiers.py -vvv\n```\n\n\n",
    "description_content_type": "text/markdown",
    "docs_url": null,
    "download_url": "",
    "downloads": {
      "last_day": -1,
      "last_month": -1,
      "last_week": -1
    },
    "home_page": "https://gitlab.developers.cam.ac.uk/uis/devops/lib/fastapi-openapierrors",
    "keywords": "",
    "license": "MIT",
    "maintainer": "",
    "maintainer_email": "",
    "name": "fastapi-openapierrors",
    "package_url": "https://pypi.org/project/fastapi-openapierrors/",
    "platform": "",
    "project_url": "https://pypi.org/project/fastapi-openapierrors/",
    "project_urls": {
      "Homepage": "https://gitlab.developers.cam.ac.uk/uis/devops/lib/fastapi-openapierrors"
    },
    "release_url": "https://pypi.org/project/fastapi-openapierrors/0.9.0/",
    "requires_dist": [
      "fastapi (>=0.68.0)"
    ],
    "requires_python": "",
    "summary": "Helpers for implementing errors nicely with OpenAPI and FastAPI.",
    "version": "0.9.0",
    "yanked": false,
    "yanked_reason": null
  },
  "last_serial": 11385520,
  "releases": {
    "0.9.0": [
      {
        "comment_text": "",
        "digests": {
          "blake2b_256": "0401b89238396b943ced9e1e39f68523b75975b2b3167584826205d94beae702",
          "md5": "357dead1799377179307bb16d61711b9",
          "sha256": "aefa0769a333cf73c406d01e5188b1c1427a7e960c440491b8d46af67964360b"
        },
        "downloads": -1,
        "filename": "fastapi_openapierrors-0.9.0-py3-none-any.whl",
        "has_sig": false,
        "md5_digest": "357dead1799377179307bb16d61711b9",
        "packagetype": "bdist_wheel",
        "python_version": "py3",
        "requires_python": null,
        "size": 3522,
        "upload_time": "2021-09-07T13:49:52",
        "upload_time_iso_8601": "2021-09-07T13:49:52.648807Z",
        "url": "https://files.pythonhosted.org/packages/04/01/b89238396b943ced9e1e39f68523b75975b2b3167584826205d94beae702/fastapi_openapierrors-0.9.0-py3-none-any.whl",
        "yanked": false,
        "yanked_reason": null
      },
      {
        "comment_text": "",
        "digests": {
          "blake2b_256": "3d496427c124678eb2044f110d7bb14dc5c7954bec6f9acf4e009448e7bddc28",
          "md5": "188ac1eb8feb0c1067ae39640a6f18a0",
          "sha256": "4a053f84ac99097c15a5bf370c407c4fdbcdefd3f5967b97cc2f2d4dec4d39da"
        },
        "downloads": -1,
        "filename": "fastapi-openapierrors-0.9.0.tar.gz",
        "has_sig": false,
        "md5_digest": "188ac1eb8feb0c1067ae39640a6f18a0",
        "packagetype": "sdist",
        "python_version": "source",
        "requires_python": null,
        "size": 3328,
        "upload_time": "2021-09-07T13:49:54",
        "upload_time_iso_8601": "2021-09-07T13:49:54.104594Z",
        "url": "https://files.pythonhosted.org/packages/3d/49/6427c124678eb2044f110d7bb14dc5c7954bec6f9acf4e009448e7bddc28/fastapi-openapierrors-0.9.0.tar.gz",
        "yanked": false,
        "yanked_reason": null
      }
    ]
  },
  "urls": [
    {
      "comment_text": "",
      "digests": {
        "blake2b_256": "0401b89238396b943ced9e1e39f68523b75975b2b3167584826205d94beae702",
        "md5": "357dead1799377179307bb16d61711b9",
        "sha256": "aefa0769a333cf73c406d01e5188b1c1427a7e960c440491b8d46af67964360b"
      },
      "downloads": -1,
      "filename": "fastapi_openapierrors-0.9.0-py3-none-any.whl",
      "has_sig": false,
      "md5_digest": "357dead1799377179307bb16d61711b9",
      "packagetype": "bdist_wheel",
      "python_version": "py3",
      "requires_python": null,
      "size": 3522,
      "upload_time": "2021-09-07T13:49:52",
      "upload_time_iso_8601": "2021-09-07T13:49:52.648807Z",
      "url": "https://files.pythonhosted.org/packages/04/01/b89238396b943ced9e1e39f68523b75975b2b3167584826205d94beae702/fastapi_openapierrors-0.9.0-py3-none-any.whl",
      "yanked": false,
      "yanked_reason": null
    },
    {
      "comment_text": "",
      "digests": {
        "blake2b_256": "3d496427c124678eb2044f110d7bb14dc5c7954bec6f9acf4e009448e7bddc28",
        "md5": "188ac1eb8feb0c1067ae39640a6f18a0",
        "sha256": "4a053f84ac99097c15a5bf370c407c4fdbcdefd3f5967b97cc2f2d4dec4d39da"
      },
      "downloads": -1,
      "filename": "fastapi-openapierrors-0.9.0.tar.gz",
      "has_sig": false,
      "md5_digest": "188ac1eb8feb0c1067ae39640a6f18a0",
      "packagetype": "sdist",
      "python_version": "source",
      "requires_python": null,
      "size": 3328,
      "upload_time": "2021-09-07T13:49:54",
      "upload_time_iso_8601": "2021-09-07T13:49:54.104594Z",
      "url": "https://files.pythonhosted.org/packages/3d/49/6427c124678eb2044f110d7bb14dc5c7954bec6f9acf4e009448e7bddc28/fastapi-openapierrors-0.9.0.tar.gz",
      "yanked": false,
      "yanked_reason": null
    }
  ],
  "vulnerabilities": []
}